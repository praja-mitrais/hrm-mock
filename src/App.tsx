import { useState } from "react";
import reactLogo from "./assets/react.svg";
import "./App.css";
import { TEXT } from "./config/env";

function App() {
  const [count, setCount] = useState(0);

  return (
    <>
      <div>
        <a href="https://react.dev">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h1>Vite + React</h1>
      <div className="card">
        <button onClick={() => setCount((prevCount) => prevCount + 1)} type="button">
          count is {count}
        </button>
        <p>
          Edit <code>src/App.tsx</code> and save to test HMR
        </p>
        {/* <p>{count > 10 ? "This is BIG" : "This is Small"}</p> */}
        <h1>{TEXT}</h1>
      </div>
      <p className="read-the-docs">Click on the Vite and React logos to learn more</p>
    </>
  );
}

export default App;
