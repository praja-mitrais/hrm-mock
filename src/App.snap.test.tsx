import * as renderer from "react-test-renderer";
import App from "./App";

jest.mock("./config/env", () => ({
  TEXT: "TEXT",
}));

it("renders correctly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
