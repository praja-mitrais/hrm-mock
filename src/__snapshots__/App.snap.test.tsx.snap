// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <div>
    <a
      href="https://react.dev"
    >
      <img
        alt="React logo"
        className="logo react"
      />
    </a>
  </div>,
  <h1>
    Vite + React
  </h1>,
  <div
    className="card"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      count is 
      0
    </button>
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to test HMR
    </p>
  </div>,
  <p
    className="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>,
]
`;
